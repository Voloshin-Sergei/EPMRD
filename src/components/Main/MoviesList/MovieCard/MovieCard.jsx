import React from 'react';
import propTypes from 'prop-types';
import style from './MovieCard.module.scss';

const MovieCard = ({
  year, title, poster, genres,
}) => (
  <div className={style.movie}>
    <img className={style.poster} src={poster} alt={title} />
    <div className={style.info}>
      <h3 className={style.title}>{title}</h3>
      <span className={style.year}>{year}</span>
    </div>
    <ul className={style.genres}>
      {genres.map((genre) => <li key={genre} className={style.genre}>{genre}</li>)}
    </ul>
  </div>
);

MovieCard.propTypes = {
  year: propTypes.string.isRequired,
  title: propTypes.string.isRequired,
  poster: propTypes.string.isRequired,
  genres: propTypes.arrayOf(propTypes.string).isRequired,
};

export default MovieCard;
