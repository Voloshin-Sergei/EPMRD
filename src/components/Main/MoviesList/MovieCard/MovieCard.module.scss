@import 'Styles/variables/colors';
@import 'Styles/variables/fonts';
@import 'Styles/variables/responsive';

.movie {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  overflow: hidden;
  color: $color-primary--darken;
  font-weight: $font-weight--light;
  background-color: $color-primary;
  border-radius: 5px;
  box-shadow: 0 4px 5px rgba(0, 0, 0, 0.2);
  cursor: pointer;

  &:hover {
    transform: translateY(-0.8rem);
  }
}

.poster {
  width: 100%;
}

.info {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem;
}

.year {
  padding: 0.3rem;
  border: 1px solid $color-primary--darken;
  border-radius: 3px;
}

.title {
  color: lighten($color-primary--dark, 20%);
  font-weight: $font-weight--light;
  font-size: 2.4rem;
}

.genres {
  display: flex;
  flex-wrap: wrap;
  padding: 0.5rem 1rem;
  list-style: none;
}

.genre {
  margin-right: 1rem;
  font-size: 1.4rem;
}
