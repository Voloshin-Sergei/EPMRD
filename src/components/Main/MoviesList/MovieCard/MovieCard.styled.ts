import styled from 'styled-components';
import { colors } from '../../../../styles/variables/colors.styled';
import { fonts } from '../../../../styles/variables/fonts.styled';

interface StyledMovieVoteProps {
  rate: string;
}

export const StyledMovie = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  overflow: hidden;
  color: ${colors.colorPrimaryDarken};
  font-weight: ${fonts.fontWeightLight};
  background-color: ${colors.colorPrimary};
  border-radius: 5px;
  box-shadow: 0 4px 5px ${colors.colorSecondaryShade};

  :hover {
    transform: translateY(-0.8rem);
  }
`;

export const StyledMoviePoster = styled.img`
  width: 100%;
`;

export const StyledMovieVote = styled.span<StyledMovieVoteProps>`
  position: absolute;
  top: 1.5rem;
  left: 1.5rem;
  padding: 0.25rem 0.5rem;
  font-weight: ${fonts.fontWeightMedium};
  font-size: 2rem;
  background-color: ${colors.colorPrimaryDarken};
  border-radius: 3px;
  color: ${(props) => props.rate};
`;

export const StyledMovieInfo = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem;
`;

export const StyledMovieYear = styled.span`
  padding: 0.3rem;
  font-size: 1.5rem;
  border: 1px solid ${colors.colorPrimaryDarken};
  border-radius: 3px;
`;

export const StyledTitle = styled.h3`
  margin-right: 2rem;
  color: ${colors.colorPrimaryDarkLighten};
  font-weight: ${fonts.fontWeightLight};
  font-size: 2.4rem;
`;

export const StyledMovieGenres = styled.ul`
  display: flex;
  flex-wrap: wrap;
  padding: 0.5rem 1rem;
  list-style: none;
`;

export const StyledMovieGenre = styled.li`
  margin-right: 1rem;
  font-size: 1.4rem;
`;
