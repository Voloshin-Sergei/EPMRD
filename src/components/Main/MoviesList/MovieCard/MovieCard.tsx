import React from 'react';
import { Link } from 'react-router-dom';
import { shortYear } from 'Helpers/shortYear';
import { setRateColorStyle } from 'Helpers/setRateColorStyle';
import { onErrorHandler } from 'Helpers/onErrorCover';
import {
  StyledMovie,
  StyledMoviePoster,
  StyledMovieVote,
  StyledMovieInfo,
  StyledTitle,
  StyledMovieYear,
  StyledMovieGenres,
  StyledMovieGenre,
} from './MovieCard.styled';

export interface MovieCardProps {
  id: number;
  year: string;
  title: string;
  poster: string;
  genres: string[];
  rating: number;
}

export const MovieCard: React.FC<MovieCardProps> = ({
  id,
  year,
  title,
  poster,
  genres,
  rating,
}) => {
  return (
    <Link to={`/movie/${id}`}>
      <StyledMovie>
        <StyledMoviePoster src={poster} alt={title} onError={onErrorHandler} />
        <StyledMovieVote rate={setRateColorStyle(rating)}>{rating}</StyledMovieVote>
        <StyledMovieInfo>
          <StyledTitle>{title}</StyledTitle>
          <StyledMovieYear>{shortYear(year)}</StyledMovieYear>
        </StyledMovieInfo>
        <StyledMovieGenres>
          {genres.map((genre: string, index) => (
            <StyledMovieGenre key={`${genre}_${index}`}>{genre}</StyledMovieGenre>
          ))}
        </StyledMovieGenres>
      </StyledMovie>
    </Link>
  );
};
