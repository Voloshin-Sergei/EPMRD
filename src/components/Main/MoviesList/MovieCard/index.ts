export { MovieCard, MovieCardProps } from './MovieCard';
