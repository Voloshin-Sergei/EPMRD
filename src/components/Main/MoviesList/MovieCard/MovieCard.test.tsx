import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { MovieCard } from './MovieCard';
Enzyme.configure({ adapter: new Adapter() });

const props = {
  year: "2021",
  title: "test title",
  poster: "test cover",
  genres: ["test genre_1", "test genres_2"]
}

describe("MovieCard component", () => {
    it('should render MovieCard component', () => {
    const component = shallow(<MovieCard {...props}/>);
    expect(component).toMatchSnapshot();
  });
});