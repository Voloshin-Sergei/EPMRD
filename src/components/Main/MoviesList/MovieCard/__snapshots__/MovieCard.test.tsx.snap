// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component should render MovieCard component 1`] = `
<div
  className="movie"
>
  <img
    alt="test title"
    className="poster"
    src="test cover"
  />
  <div
    className="info"
  >
    <h3
      className="title"
    >
      test title
    </h3>
    <span
      className="year"
    >
      2021
    </span>
  </div>
  <ul
    className="genres"
  >
    <li
      className="genre"
      key="test genre_1_0"
    >
      test genre_1
    </li>
    <li
      className="genre"
      key="test genres_2_1"
    >
      test genres_2
    </li>
  </ul>
</div>
`;
